source/server/hot_restarting_base.cc

```diff
@@ -0,0 +1,118 @@
+#include "server/hot_restarting_base.h"
+
+#include "common/api/os_sys_calls_impl.h"
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Server {
+
+void HotRestartingBase::initDomainSocketAddress(sockaddr_un* address) {
+  memset(address, 0, sizeof(*address));
+  address->sun_family = AF_UNIX;
+}
+
+sockaddr_un HotRestartingBase::createDomainSocketAddress(uint64_t id, const std::string& role) {
+  // Right now we only allow a maximum of 3 concurrent envoy processes to be running. When the third
+  // starts up it will kill the oldest parent.
+  const uint64_t MAX_CONCURRENT_PROCESSES = 3;
+  id = id % MAX_CONCURRENT_PROCESSES;
+
+  // This creates an anonymous domain socket name (where the first byte of the name of \0).
+  sockaddr_un address;
+  initDomainSocketAddress(&address);
+  StringUtil::strlcpy(&address.sun_path[1],
+                      fmt::format("envoy_domain_socket_{}_{}", role, base_id_ + id).c_str(),
```

Comment:
The one functional change. Since each HotRestarting{Child,Parent} has one of these, and we have one of each, we now have two datagram domain sockets: one for talking to the parent, one for eventually talking to the child. Their sockaddr_un addresses are now differentiated: they include the word "child" if the socket with that address is being used by the child half of the logic, "parent" otherwise.